lib/ex_doc/formatter/html.ex

```diff
@@ -79,6 +83,113 @@ defmodule ExDoc.Formatter.HTML do
     end
   end
 
+  defp group_by_nesting([], _), do: []
```

Comment:
What if we move this to the `group_matcher`? This way we can unit test it too.

Reply:
That makes sense. I put it here as it was mainly and HTML-related concern, but conceptually it fits within the `GroupMatcher`, especially as it enables unit testing.